src/UI/Layout.hs

```diff
@@ -25,6 +29,12 @@ offset by = wrap . Offset by
 resizeable :: (Size (Maybe a) -> Layout a b) -> Layout a b
 resizeable = wrap . Resizeable
 
+getMaxSize :: Layout a (Size (Maybe a))
+getMaxSize = wrap (Resizeable pure)
```

Comment:
This is a pretty fun trick: `resizeable` is in continuation-passing style, `getMaxSize` is in imperative style, and they use the same constructor.